### DISCUSSION TITLE
Is there a way to restart hyprland without getting kicked to the TTY / login manager?

### DISCUSSION THREAD

**Discussion by BenMac31** [hyprwm/Hyprland#1849] [Q&A]
I need to restart hyprland to disable my GPU completely to preserve battery life. Going back to the login manager every time is a pain, so I'd ideally want to restart hyprland as quickly as possible. And if possible maybe preserve a couple of applications.

**Comment by eriteric** [comment]
You can run `hyprctl reload` to reload the config to make any changes take effect and reload the monitor. As far as actually restarting, I'm not sure.

**Reply by johanndominicthomas** [reply]
After changing the wallpaper ,using ```hyprctl reload``` does'nt change the wallpaper. Is there a way for that. My main purpose is to see the wallpaper change without having to relogin again

**Reply by tannerellen** [reply]
> After changing the wallpaper ,using `hyprctl reload` does'nt change the wallpaper. Is there a way for that. My main purpose is to see the wallpaper change without having to relogin again

How are you loading the wallpaper? You should use hyprpaper if that's not what you are using then you should read the docs: https://wiki.hypr.land/Hypr-Ecosystem/hyprpaper/

Then you can load a wallpaper and have it shown immediately with `hyprctl hyprpaper reload ,"~/your-wallpaper.png"`

You'll also want to make sure you disable the default anime girl wallpaper if you are setting custom wallpapers. Check out "disable_hyprland_logo" here: https://wiki.hypr.land/Configuring/Variables/ You will want to set that variable to true so it doesn't set before you have a chance to set your custom wallpaper as that could show a brief flash of the default anime girl wallpaper before yours is set.

**Comment by da-x** [comment]
Originally it was possible with X11 Window Manager, e.g restart the manager without losing the windows.

But with the Wayland compositor architecture it is more difficult, as the compositor process keeps much of the state, it cannot re-execute itself. However with a shared library load/unload architecture it may be possible.

I attached a plan.

[hot-reload-library.md](https://github.com/user-attachments/files/29057186/hot-reload-library.md)

@vaxerski what do you think about this? Is it worth it? Would it make debugging stuff faster?
